### DISCUSSION TITLE
Which performance is better for alluxio and juicefs?

### DISCUSSION THREAD

**Discussion by jinhang** [juicedata/juicefs#42] [Q&A] [answered]
maybe everyone wants to know this and make a choice

**Comment by davies** [comment, marked as answer]
We have done an benchmark to compare Alluxio, JindoFS and JuiceFS, the details is here: https://juicefs.com/blog/cn/posts/cache-system-of-hadoop-comparison/

Overall, they have similar performance when there is enough caching space (memory or SSD). But when the caching space is limited (<20% of dataset), JuiceFS is much faster than the other two. Because that  the cached block is much smaller than the other two (4M < 128M), is better for random read (parquet and ORC) workload.

**Reply by jinhang** [reply]
yes,I got it, Thanks for your reply for me.  juicefs look great. Is the current open source all the code? What is the difference between version v0.9.1 and version JuiceFS 4.4.9  on the official website (https://juicefs.com/docs/zh/release.html)? from github relese can not get anything.
Looking forward to your reply

**Reply by davies** [reply]
The open source community version has most of the source code in SaaS version 4.4.9, except the client that talk to our internal SaaS meta server.

**Reply by davies** [reply]
The Java SDK used to big data workload will also be open sourced in community version.
